### DISCUSSION TITLE
Activator.CreateInstance returning object? instead of object.

### DISCUSSION THREAD

**Discussion by sjb-sjb** [dotnet/runtime#45293] [Q&A] [answered]
According to the following post, Activator.CreateInstance should return object instead of object?. But in .NET 5 it is marked as returning object?. 

Post: https://stackoverflow.com/questions/8691601/creating-a-nullable-object-via-activator-createinstance-returns-null

**Comment by svick** [comment, marked as answer]
I don't understand, that question complains that `Activator.CreateInstance` *does* return `null` in some cases (namely, for nullable value types). So I think `object?` is the correct return type.

**Reply by yaakov-h** [reply]
Yep, that seems correct to me. Using C# Interactive:

```
> Activator.CreateInstance<int?>()
null
```

**Reply by gfoidl** [reply]
Also see https://github.com/dotnet/runtime/issues/674 for the same question.

**Comment by sjb-sjb** [comment]
When I read the accepted answer in the post that I referenced, I thought it meant that the fix got rid of the null return. But it seems that is not the case. I guess the point is that you cannot return an object referring to a Nullable&lt;T&gt; due to the boxing behavior of Nullable. I would have thought it would throw InvalidOperationException in this situation...

Thanks for the clarification.
